modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/jdbc/JdbcMetadataCatalog.java

```diff
@@ -182,28 +202,24 @@ public CompletableFuture<Collection<String>> getSchemasMeta(String schemaNamePtr
             schemas.add(DEFAULT_SCHEMA_NAME);
         }
 
-        return tables.tablesAsync().thenApply(tablesList ->
-                tablesList.stream()
+        return tablesAtNow().thenApply(tables ->
+                tables.stream()
                     .map(tbl -> DEFAULT_SCHEMA_NAME)
                     .filter(schema -> matches(schema, schemaNameRegex))
-                    .collect(Collectors.toCollection(() -> schemas))
+                    .collect(toCollection(() -> schemas))
         );
     }
 
     /**
-     * Creates primary key metadata from table object.
+     * Creates primary key metadata from a table descriptor.
      *
      * @param tbl Table.
      * @return Jdbc primary key metadata.
      */
-    private JdbcPrimaryKeyMeta createPrimaryKeyMeta(Table tbl) {
-        final String keyName = PK + tbl.name();
-
-        SchemaRegistry registry = ((TableImpl) tbl).schemaView();
+    private JdbcPrimaryKeyMeta createPrimaryKeyMeta(CatalogTableDescriptor tbl) {
+        String keyName = PK + tbl.name();
```

Comment:
Unrelated question: what happens after table rename? Do we lose the ability to access the PK index?

Reply:
Probably, when we rename a table, we'll also (atomically) rename the corresponding PK. Should I file an issue?

Reply:
I don't know, we should discuss this behavior with someone else too